com.ca.lsp.cobol/lsp-core-cobol-parser/src/test/java/com/ca/lsp/core/cobol/preprocessor/sub/line/rewriter/impl/CobolLineIndicatorProcessorImplTest.java

```diff
@@ -0,0 +1,169 @@
+    continuationLineWithEndQuotes.setContentAreaA("       \"RANDOM TEXT \"  ");
+    continuationLineWithEndQuotes.setContentAreaB("        ");
+
+    final List<CobolLine> listOfLines =
+        Arrays.asList(
+            goodContinuationLineToTest,
+            singleContinuationLineToTest,
+            trailingQuoteContinuationLine,
+            continuationLine,
+            badContinuationLineToTest,
+            emptyContinuationLineToTest,
+            commentLineToTest,
+            compilerDirectiveLineToTest,
+            normalLineToTest,
+            debugLineToTest,
+            continuationLineWithEndQuotes);
+    CobolLineIndicatorProcessorImpl processor = new CobolLineIndicatorProcessorImpl();
+    processor.processLines(listOfLines);
+
+    assertEquals(
+        WS + "\"RANDOM TEXT           ",
+        processor.processLines(listOfLines).get(0).getIndicatorArea()
+            + processor.processLines(listOfLines).get(0).getContentArea());
+    assertEquals(
+        WS + "'RANDOM TEXT           ",
+        processor.processLines(listOfLines).get(1).getIndicatorArea()
+            + processor.processLines(listOfLines).get(1).getContentArea());
+    assertEquals(
+        WS + "\"",
+        processor.processLines(listOfLines).get(2).getIndicatorArea()
+            + processor.processLines(listOfLines).get(2).getContentArea());
+    assertEquals(
+        WS + "CONTINUED LINE ENDS HERE\"",
+        processor.processLines(listOfLines).get(3).getIndicatorArea()
+            + processor.processLines(listOfLines).get(3).getContentArea());
+    assertEquals(
+        WS + "RANDOM TEXT",
+        processor.processLines(listOfLines).get(4).getIndicatorArea()
+            + processor.processLines(listOfLines).get(4).getContentArea());
+    assertEquals(
+        WS + EMPTY_STRING,
+        processor.processLines(listOfLines).get(5).getIndicatorArea()
+            + processor.processLines(listOfLines).get(5).getContentArea());
+    assertEquals(
+        COMMENT_TAG + WS + "THIS IS A COMMENT",
+        processor.processLines(listOfLines).get(6).getIndicatorArea()
+            + processor.processLines(listOfLines).get(6).getContentArea());
+    assertEquals(
+        WS + EMPTY_STRING,
+        processor.processLines(listOfLines).get(7).getIndicatorArea()
+            + processor.processLines(listOfLines).get(7).getContentArea());
+    assertEquals(
+        WS + "         RANDOM TEXT , ",
+        processor.processLines(listOfLines).get(8).getIndicatorArea()
+            + processor.processLines(listOfLines).get(8).getContentArea());
+    assertEquals(
+        WS + "     DEBUG LINE HERE",
+        processor.processLines(listOfLines).get(9).getIndicatorArea()
+            + processor.processLines(listOfLines).get(9).getContentArea());
+    assertEquals(
+        WS + "RANDOM TEXT \"",
+        processor.processLines(listOfLines).get(10).getIndicatorArea()
+            + processor.processLines(listOfLines).get(10).getContentArea());
```

Comment:
We should add a line of comment (or a javadoc if defined in single unit test) to explain in detail what we are trying to verify.

Reply:
reworked the junit test and added proper javadoc information for each test class